src/internal/compilerSupport/v2.ts

```diff
@@ -320,22 +320,22 @@ export function callFunction(rawFnArgs: Array<any>, def: FunctionCallDef, script
             };
         });
 
-        // registering handler for particle timeout
-        peer.internals.regHandler.timeout(particle.id, () => {
-            reject(`Request timed out for ${def.functionName}`);
+        peer.internals.initiateParticle(particle, (stage) => {
+            if (def.returnType.tag === 'void' && (stage.stage === 'sent' || stage.stage === 'localWorkDone')) {
+                resolve(undefined);
+            }
```

Comment:
Awesome! Let's add a comment.

```suggestion
            // If function is void, then it's completed when one of the two conditions is met:
            //  1. The particle is sent to the network (state 'sent')
            //  2. All CallRequests are executed, e.g., all variable loading and local function calls are completed (state 'localWorkDone')
            if (def.returnType.tag === 'void' && (stage.stage === 'sent' || stage.stage === 'localWorkDone')) {
                resolve(undefined);
            }
```